File: src/main/java/jp/developer/bbee/pcassem/presentation/data/ReviewResponse.java
package jp.developer.bbee.pcassem.presentation.data;

public class ReviewResponse {
    public ReviewSuccess successData = null;
    public ReviewFailure failureData = null;

    public ReviewResponse(ReviewSuccess data) {
        this.successData = data;
    }

    public ReviewResponse(ReviewFailure data) {
        this.failureData = data;
    }
}

File: src/main/java/jp/developer/bbee/pcassem/presentation/data/ReviewResponse.kt
package jp.developer.bbee.pcassem.presentation.data

class ReviewResponse private constructor() {
    @JvmField var successData: ReviewSuccess? = null
    @JvmField var failureData: ReviewFailure? = null

    constructor(data: ReviewSuccess) : this() { successData = data }
    constructor(data: ReviewFailure) : this() { failureData = data }
}
